src/lustre/lustreSyntaxChecks.ml

```diff
@@ -531,7 +531,8 @@ let rec expr_only_supported_in_merge observer expr =
   | ConvOp (_, _, e)
   | Pre (_, e)
   | Current (_, e)
-  | Quantifier (_, _, _, e) -> r observer e
+  | Quantifier (_, _, _, e) 
+  | ChooseOp (_, _, e) -> r observer e
```

Comment:
You should reset observer to `false` for the `ChooseOp` case, i.e., use the rule: `| ChooseOp (_, _, e) -> r false e` and restore the rule for the other cases. Even if the "node call" of the choose operator occurs in the argument of a `merge` operator, the predicate `e` should not contain any `when` or `activate` expression unless they occur in the argument of a `merge` operator.